mm/dynamic/c/Magazine.c

```
@@ -168,11 +249,28 @@ void MagazineFree(void* ptr) {
     // --- Determine Slab and Size Class (lockless via page-hash) ---
     Slab* slab = FindSlabForPointer(ptr);
     if (!slab) {
-        // Assume it's a large allocation
+        // Not found in any slab: either a magazine large allocation or foreign (e.g., Rust) allocation in HYBRID mode
         LargeBlockHeader* header = (LargeBlockHeader*)ptr - 1; // Get pointer to header
-        // Basic sanity: header must be mapped; we trust VMem
-        VMemFree((void*)header, header->size + sizeof(LargeBlockHeader));
+        if (header && header->magic == LARGE_BLOCK_MAGIC) {
+            // Magazine large allocation
+            if (g_validation_level != KHEAP_VALIDATION_NONE) {
+                FastMemset(ptr, 0xDD, header->size);
```

Clang (14.0.6)

[warning] 0xDD is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)